wizeline/project-lab
# Feat - #226 - Assign owner on leave

#### What does this PR do?

When a project member that is the OWNER wants to leave the project, before he is able to do so, he is prompted to selected a new owner from within the active members.

#### Where should the reviewer start?

- app/pages/projects/[projectId]/index.tsx
- app/pages/projects/[projectId]/ProjectConfirmationModal.tsx
- app/projects/mutations/updateProjectOwner.ts

#### How should this be manually tested?

(1st approach)
1. Run project locally
2. Log in
3. Create a brand new proposal (+ New proposal)
4. Once created, you should be automatically the Owner. Try to leave project (click on 'Leave Project')
5. You should be prompted to select a new Owner. In this case as there is no other member, you will not be able to leave the project.

(2nd approach)
1. Run project locally
2. Log in
3. Select whichever project you want and Join it.
4. Now you need to become the Owner, so will need to edit the DB directly. DB -> 'Projects' table -> 'ownerId' field, input your profileId. Save it
5. Go back to the project page, refresh, you should be the owner. Try to 'Leave project'.
7. You should be prompted to select a new Owner. Choose one and Confirm.
8. You should now be Inactive and a new Owner should have been assigned.

#### Any background context you want to provide?

N/A

#### What are the relevant tickets?

#226 

#### Screenshots

<img width="522" alt="image" src="https://user-images.githubusercontent.com/97301313/156839270-b5956906-ec65-46b6-9273-b24bbcbffd73.png">

#### Questions

N/A

#### Checklist
